core/src/main/java/org/apache/iceberg/io/ResolvingFileIO.java

```diff
@@ -95,25 +100,12 @@ public void deleteFiles(Iterable<String> pathsToDelete) throws BulkDeletionFailu
     Iterators.partition(pathsToDelete.iterator(), BATCH_SIZE)
         .forEachRemaining(
             partitioned -> {
-              Map<FileIO, List<String>> pathByFileIO =
+              Map<DelegateFileIO, List<String>> pathByFileIO =
```

Comment:
We can just call the delegate directly now, I don't believe there is a need for this logic anymore

Reply:
do you think we should still batch here or would we want to essentially use the same implementation you had?
```
// peek at the first element to determine the type of FileIO
    Iterator<String> originalIterator = pathsToDelete.iterator();
    if (!originalIterator.hasNext()) {
      return;
    }

    PeekingIterator<String> iterator = Iterators.peekingIterator(originalIterator);
    DelegateFileIO fileIO = io(iterator.peek());
    fileIO.deleteFiles(() -> iterator);
```

To me it seems like batching should probably be handled by the underlying FileIO implementation, but just wanted to see what others think?

Reply:
actually right, it is probably better to keep this as there may be multiple FileIO types.